spec.go

```diff
@@ -448,15 +448,48 @@ func createLibcontainerMount(cwd, dest string, m specs.Mount) *configs.Mount {
 	}
 }
 
+// pathClean makes a path safe for use with filepath.Join. This is done by not
+// only cleaning the path, but also (if the path is relative) adding a leading
+// '/' and cleaning it (then removing the leading '/'). This ensures that a
+// path resulting from prepending another path will always resolve to lexically
+// be a subdirectory of the prefixed path. This is all done lexically, so paths
+// that include symlinks won't be safe as a result of using pathClean.
+func pathClean(path string) string {
```

Comment:
Ah, okay. I see what you've done. I'm not too sure why you've moved this. Docker binds directly into the `libcontainer` stuff, so this change will break security **in Docker**.